src/lib.rs

```diff
@@ -277,28 +289,44 @@ pub trait PrimeField: Field + From<u64> {
     /// This is usually `Self::NUM_BITS - 1`.
     const CAPACITY: u32;
 
-    /// Returns a fixed multiplicative generator of `modulus - 1` order. This element must
-    /// also be a quadratic nonresidue.
+    /// A fixed multiplicative generator of `modulus - 1` order. This element must also be
+    /// a quadratic nonresidue.
     ///
     /// It can be calculated using [SageMath] as `GF(modulus).primitive_element()`.
     ///
-    /// Implementations of this method MUST ensure that this is the generator used to
+    /// Implementations of this trait MUST ensure that this is the generator used to
     /// derive `Self::root_of_unity`.
     ///
     /// [SageMath]: https://www.sagemath.org/
-    fn multiplicative_generator() -> Self;
+    const MULTIPLICATIVE_GENERATOR: Self;
+
+    /// Returns a fixed multiplicative generator of `modulus - 1` order. This element must
+    /// also be a quadratic nonresidue.
+    ///
+    /// Implementations of this method MUST ensure that this is the generator used to
+    /// derive `Self::root_of_unity`.
+    #[inline(always)]
+    fn multiplicative_generator() -> Self {
+        Self::MULTIPLICATIVE_GENERATOR
+    }
```

Comment:
Can we remove this method entirely (and the others like it) and just have end users use the associated consts?

Reply:
Yep, I cannot think of any case where someone could use a trait method but not a trait associated constant.